pandas/core/indexes/numeric.py

```diff
@@ -174,6 +175,12 @@ class Int64Index(NumericIndex):
     _engine_type = libindex.Int64Engine
     _default_dtype = np.int64
 
+    def __contains__(self, key):
+        hash(key)
```

Comment:
can you add a doc-string

Reply:
can you create a IntegerIndex(NumericIndex) and put this method there, then inherit Int64Index and UInt64Index from that?